lib/rubocop/cop/rspec/multiple_expectations.rb

```diff
@@ -50,14 +50,23 @@ class MultipleExpectations < Cop
 
         MSG = 'Example has too many expectations [%<total>d/%<max>d].'
 
-        def_node_search :with_aggregate_failures?, '(sym :aggregate_failures)'
-        def_node_search :disabled_aggregate_failures?, <<-PATTERN
-          (pair (sym :aggregate_failures) (false))
+        def_node_matcher :aggregate_failures?, <<-PATTERN
+          (block {
+              (send _ _ <(sym :aggregate_failures) ...>)
+              (send _ _ ... (hash <(pair (sym :aggregate_failures) true) ...>))
+            } ...)
+        PATTERN
+
+        def_node_matcher :aggregate_failures_present?, <<-PATTERN
+          (block {
+              (send _ _ <(sym :aggregate_failures) ...>)
+              (send _ _ ... (hash <(pair (sym :aggregate_failures) _) ...>))
+            } ...)
```

Comment:
Appreciate a hint how to collapse the two above node patterns into one with [`%` syntax](https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/node_pattern.rb#L69).

Tried replacing them with:
```ruby
        def_node_matcher :aggregate_failures?, <<-PATTERN
          (block {
              (send _ _ <(sym :aggregate_failures) ...>)
              (send _ _ ... (hash <(pair (sym :aggregate_failures) %) ...>))
            } ...)
        PATTERN
```

and using as `aggregate_failures?(node, true)`, `aggregate_failures?(node, :true)`, and `aggregate_failures?(node, 'true')` - none of those variants worked.

Reply:
How about just capturing the argument, then in the block that checks for aggregate_failures you just need to check whether the node matched and if there is a captured value - it is true?

Reply:
It turns out to become quite hacky for two reasons:
 - I have to also capture from the other `{}` branch
 - in the latter branch it's expected to be a boolean, for the former it will be a symbol

so I have to check it like that:
```ruby
          aggregate_failures = aggregate_failures?(node_with_aggregate_failures)
          aggregate_failures.true_type? || aggregate_failures.sym_type?
```

it works, but it's hard to understand.
Tried reading RuboCop's own usages of `%`-syntax, still can't understand how it works, and if it's possible to pass it an `_`.

I'd rather keep this as is, it's easier to spot the redundancy like that and address in the future. WDYT?


On the other note, I added a line to the changelog, but there were no user-facing changes, so just mentioning that RuboCop version dependency has been updated.

Reply:
@marcandre Do you happen to know if it's possible to pass a boolean or an `any` as a node matcher argument (to substitute `%`)? I think I tried with booleans and failed.

Reply:
Super interesting question. Turns out you won't be able to because the check is that `node == your_arg`. We should revert that order for more flexibility, so it is possible to define an any matcher (`ANY = Object.new.tap {|o| def o.==(_); true; end}`)

Reply:
Got it, thanks!
Please keep in mind it's a pretty narrow case with just one potential usage I know if. Other cops here in `rubocop-rspec` using `%` provide symbols, and it's quite sufficient for them.

And this cop works as well, I just keep it somewhere in my mind's dirty corner that it might be improved.

Reply:
It's a simple change, but so much more powerful `https://github.com/rubocop-hq/rubocop-ast/pull/31`. You could have `ANY = ->(_) { true }` and pass that.

Reply:
Two node patterns easily collapsed into one with this new generic parameters https://github.com/rubocop-hq/rubocop-rspec/pull/930